gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/storage/AWSStorageService.java

```diff
@@ -4,87 +4,117 @@
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.model.DeleteObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.presigner.S3Presigner;
+import software.amazon.awssdk.services.s3.presigner.model.GetObjectPresignRequest;
+
+@Slf4j
+@SuppressFBWarnings(value = "EI_EXPOSE_REP2", justification = "S3Client is immutable and thread-safe.")
 public class AWSStorageService implements StorageService {
 
-    private static final long SIXY_MINUTES = 1000 * 60 * 60;
-    private final AmazonS3 s3Client;
+    private static final long SIXTY_MINUTES = 60;
+    private final S3Client s3Client;
     private final String bucketName;
+    private final S3Presigner s3Presigner;
 
-    public AWSStorageService(StorageServiceConfiguration configuration) {
-
-        var clientBuilder = AmazonS3ClientBuilder.standard();
+    public AWSStorageService(S3Client s3Client,
+                             StorageServiceConfiguration configuration,
+                             S3Presigner.Builder presignerBuilder) {
 
+        AwsBasicCredentials credentials = null;
+        StaticCredentialsProvider credentialsProvider = null;
         if (accessKeyProvided(configuration)) {
+            credentials = AwsBasicCredentials.create(configuration.getAccountReference(), configuration.getAccountSecret());
+            credentialsProvider = StaticCredentialsProvider.create(credentials);
+        }
 
-            AWSCredentials credentials = new BasicAWSCredentials(
-                configuration.getAccountReference(),
-                configuration.getAccountSecret()
-            );
+        this.s3Client = (credentialsProvider == null)
+                        ? s3Client : S3Client.builder()
+                                             .region(Region.of(configuration.getRegion()))
+                                             .credentialsProvider(credentialsProvider)
+                                             .build();
 
-            clientBuilder
-                .withCredentials(new AWSStaticCredentialsProvider(credentials))
-                .withRegion(configuration.getRegion());
-        }
+        this.s3Presigner = presignerBuilder
+                .region(Region.of(configuration.getRegion()))
```

Comment:
What happens when the region isn't specified in the configuration? Does this cause this to blow up?